cmd/controller/main.go

```diff
@@ -15,15 +15,30 @@ limitations under the License.
 package main
 
 import (
-	"context"
+	"fmt"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 
 	"github.com/aws/karpenter/pkg/cloudprovider"
 	"github.com/aws/karpenter/pkg/cloudprovider/aws"
+	cloudprovidermetrics "github.com/aws/karpenter/pkg/cloudprovider/metrics"
 	"github.com/aws/karpenter/pkg/controllers"
+	"github.com/aws/karpenter/pkg/sharedmain"
 )
 
+const componentName = "controller"
+
 func main() {
-	controllers.Initialize(func(ctx context.Context, options cloudprovider.Options) cloudprovider.CloudProvider {
-		return aws.NewCloudProvider(ctx, options)
-	})
+	options, manager := sharedmain.NewOptionsWithManagerOrDie(componentName)
+	cloudProvider := cloudprovider.CloudProvider(aws.NewCloudProvider(options.Ctx, cloudprovider.Options{Options: options}))
+	if hp, ok := cloudProvider.(sharedmain.HealthCheck); ok {
+		utilruntime.Must(manager.AddHealthzCheck("cloud-provider", hp.LivenessProbe))
+	}
+	cloudprovidermetrics.Decorate(cloudProvider)
```

Comment:
IIUC this doesn't actually decorate the cloudprovider, since the return value is the decorated type. We need tests for our metrics.

Works if you inline it on line 40, or reassign here.